### PR TITLE
fix-32253

### PR DESCRIPTION
**Pull request by teja2**

### CODE REVIEW

File: src/cdk/table/table.ts [hunk 1]
```diff
@@ -1330,6 +1331,7 @@ export class CdkTable<T>
    * `multiTemplateDataRows` or adding/removing row definitions.
    */
   private _forceRenderDataRows() {
+    this._viewRepeater.detach();
```
Comment (crisbeto) [inline src/cdk/table/table.ts]: It looks like this breaks some tests. Also my impression is that `detach` is only meant to be called once.

**Comment by teja2** [comment]
@crisbeto updated the PR. Added a clearCache hook to _RecycleViewRepeaterStrategy and switched CdkTable’s forced re-render path to call it so recycled rows are flushed without repeatedly tearing down the entire repeater.

**Comment by teja2** [comment]
Fixed the tests as well

**Comment by crisbeto** [comment]
It looks like the tests are still failing. Also CLA isn't signed.

**Comment by teja2** [comment]
@crisbeto can you please now. locally it is passing

**Comment by ssc-io** [comment]
Context from another team hitting #32253 on `@angular/cdk` 22.0.5: this is broader than opt-in
`recycleRows`, which may be worth weighing against the current P3.

Inside a `cdk-virtual-scroll-viewport` the recycling repeater is selected unconditionally
(`this.recycleRows || this._virtualScrollEnabled()`), so every virtual-scrolled `cdk-table` with
toggleable or reorderable columns is affected without ever using `recycleRows` — and consumers cannot
opt out, since the input only turns recycling *on*. There the cache also fills without any data
change: scrolling down and back is enough. Reordering columns fails the same way as hiding/showing —
the header updates, recycled rows keep their old cells.

Mechanism and a minimal repro without virtual scrolling are in #32253, and they support this PR's
approach: `_forceRenderDataRows()` leaves `_viewCache` intact, cached views come back through
`_insertView()` as `REPLACED`, and `renderRows()` stamps cell templates only for `INSERTED`.

`main` still carries the unpatched `_forceRenderDataRows()`. Until this lands, the only handle
consumers have is calling `detach()` on the private `_viewRepeater` whenever the columns change.
